### PR TITLE
Optimism Gateway v1

### PR DESCRIPTION
**Pull request by gigamesh**
- Adds Welcome modal that will eventually contain disclaimer text about the dangers of aping into Optimism
- Lifts a bunch of app state to App.tsx (ex: walllet provider)
- General UI improvements
- Adds blocknative's Notify.js -- only useable for deposits now but I asked if they can add our chain IDs so we can at least get revert messages for failed withdrawals
- Refactored subgraph queries to filter by address when one is present
- Adds ability to query kovan subgraphs

### CODE REVIEW

File: packages/react-app/src/App.tsx [hunk 1]
```diff
@@ -1,48 +1,501 @@
...
+      toastIdRef.current = toast({
+        title: 'Error',
+        position: 'bottom-right',
+        description: (
+          <Text>
+            {message || (
+              <>
+                Please try again. If you continue to experience problems, please reach out to us on{' '}
+                <Link
+                  href="https://www.twitch.tv/optimismpbc"
```
Comment (K-Ho) [inline packages/react-app/src/App.tsx:67]: wrong link

File: packages/react-app/src/App.tsx [hunk 2]
```diff
@@ -1,48 +1,501 @@
...
+    const l1EthGateway = tokenList.find(
+      tokenData =>
+        tokenData.symbol === 'ETH' &&
+        (tokenData.chainId === chainIds.KOVAN_L1 || tokenData.chainId === chainIds.MAINNET_L1)
+    );
+    const l2EthGateway = tokenList.find(
+      tokenData =>
+        tokenData.symbol === 'ETH' &&
+        (tokenData.chainId === chainIds.KOVAN_L2 || tokenData.chainId === chainIds.MAINNET_L2)
+    );
```
Comment (K-Ho) [inline packages/react-app/src/App.tsx:124]: Can probably make a helper and generalize this for supporting other tokens

File: packages/react-app/src/App.tsx [hunk 3]
```diff
@@ -1,48 +1,501 @@
...
+        },
+        l2: {
+          ethGateway: new Contract(l2EthGatewayAddr, abis.l2.standardBridge, rpcL2),
+        },
+      };
+
+      setContracts(contracts);
+      setConnectedChainId(chainId);
+      setWalletProvider(provider);
+      setRpcL1Provider(rpcL1);
```
Comment (K-Ho) [inline packages/react-app/src/App.tsx:145]: Is this needed? We have a `getRpcProviders function below`

**Review by gigamesh — COMMENTED** [review]
- [x ] Look at `setRpcL1Provider` and `getRpcProviders` to see if there is a better pattern (kind of confusing to have both)

**Review by K-Ho — APPROVED** [review]
LGTM!

**Review by snario — APPROVED** [review]

**Pull request merged**